internal/services/automanage/automanage_configuration_profile_resource.go

```diff
@@ -0,0 +1,228 @@
+			DiffSuppressFunc: pluginsdk.SuppressJsonDiff,
+		},
+
+		"tags": commonschema.Tags(),
+	}
+}
+
+func (r AutoManageConfigurationProfileResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (r AutoManageConfigurationProfileResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var model AutoManageConfigurationProfileModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			client := metadata.Client.Automanage.ConfigurationProfileClient
+			subscriptionId := metadata.Client.Account.SubscriptionId
+			id := parse.NewAutomanageConfigurationProfileID(subscriptionId, model.ResourceGroupName, model.Name)
+			existing, err := client.Get(ctx, id.ConfigurationProfileName, id.ResourceGroup)
+			if err != nil && !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("checking for existing %s: %+v", id, err)
+			}
+
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			properties := automanage.ConfigurationProfile{
+				Location:   utils.String(location.Normalize(model.Location)),
+				Properties: &automanage.ConfigurationProfileProperties{},
+				Tags:       tags.FromTypedObject(model.Tags),
+			}
+
+			if model.Configuration != "" {
+				var configurationValue interface{}
+				err = json.Unmarshal([]byte(model.Configuration), &configurationValue)
+				if err != nil {
+					return err
+				}
+				properties.Properties.Configuration = &configurationValue
+			}
+
+			if _, err := client.CreateOrUpdate(ctx, id.ConfigurationProfileName, id.ResourceGroup, properties); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (r AutoManageConfigurationProfileResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Automanage.ConfigurationProfileClient
+
+			id, err := parse.AutomanageConfigurationProfileID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			var model AutoManageConfigurationProfileModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			resp, err := client.Get(ctx, id.ConfigurationProfileName, id.ResourceGroup)
+			if err != nil {
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			if metadata.ResourceData.HasChange("configuration_json") {
+				var configurationValue interface{}
+				err := json.Unmarshal([]byte(model.Configuration), &configurationValue)
+				if err != nil {
+					return err
+				}
+
+				resp.Properties.Configuration = &configurationValue
+			}
+
+			resp.SystemData = nil
```

Comment:
The ARM spec specifies that a service is suppose to ignore read-only fields on updates so this shouldn't need to set to nil - if it's required then this would indicated a bug in the API?

Reply:
I will take a look at this today and change related code.